.github/workflows/publish.yml

```diff
@@ -1,24 +1,27 @@
 name: Publish packages on NPM 📦
+
 on:
   push:
     branches:
       - main
+
 jobs:
   publish:
+    timeout-minutes: 20
+
     runs-on: ubuntu-latest
+
     env:
       NODE_ENV: test
       DATABASE_URL: postgres://
       AUTH_SECRET: test
+
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
-        with:
-          node-version: 16
-          registry-url: "https://registry.npmjs.org"
-          cache: "yarn"
-        env:
-          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - uses: ./.github/actions/ci-setup
+
       - run: yarn
       - run: yarn turbo run build
       - run: yarn turbo run publish-to-npm
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
In a way, this reverts the change from [this commit](https://github.com/webstudio-is/webstudio-designer/commit/44e5c8bc77c6b8dd60dec7bc2d7c8f099cbdac0a). I think that this should be fine but @TrySound has changed this for a reason - so perhaps there is something that I'm missing here. 

I'd like to avoid passing this token to the `ci-setup` step as that would make this token available to all workflows. From the security PoV it's better to scope this as much as possible - and we can't scope it much further than a single step :p

Reply:
A notable difference between what I have here and what you had there before is that I've removed `registry-url` input from `actions/setup-node`. I don't think it's needed as that's the default registry anyway. And since I don't use this input the action itself no longer creates `.npmrc` as [this whole thing](https://github.com/actions/setup-node/blob/41acaa2e85a9c20b0bcaff1397645d5f33cab6ac/src/authutil.ts) gets skipped (this `.npmrc` shouldn't be required here anyway).

Reply:
All is needed. I tried all possible combinations. registry-url is needed and token no setup-node is needed.

Reply:
This still baffles me a little - I maintain Changesets, a tool responsible for versioning and publishing packages. To the best of my knowledge, what I have in this PR should work. However, there are some nuances, and it's easy to miss things. It's also not the part of the codebase that I often revisit/touch so it's not like I exactly know all of the details.

I'll try to do a few test runs in a separate kitchen sink project to test it out and get a better feeling of the problem.